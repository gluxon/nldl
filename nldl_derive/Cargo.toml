[package]
name = "nldl_derive"
version = "0.1.0"
authors = ["Brandon Cheng <brandon.cheng@protonmail.com>"]
edition = "2021"

[lib]
proc-macro = true

[dependencies]
syn = "1.0"
proc-macro2 = "1.0.18"
quote = "1.0"
thiserror = "1.0"

[dev-dependencies]
nldl = { path = "../nldl" }
trybuild = "1.0"
