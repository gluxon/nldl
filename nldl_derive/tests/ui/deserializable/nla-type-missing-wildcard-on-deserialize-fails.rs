use nldl_derive::NetlinkAttributeDeserializable;

#[derive(Debug, PartialEq, NetlinkAttributeDeserializable)]
#[netlink15(deserialize(error = "ParseNlaIntError"))]
enum ControllerAttributeOperation {
    #[nla_type(0)]
    Unspec,
}

fn main() {}
