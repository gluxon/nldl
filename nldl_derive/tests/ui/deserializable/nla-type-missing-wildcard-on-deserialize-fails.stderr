error: proc-macro derive panicked
 --> tests/ui/deserializable/nla-type-missing-wildcard-on-deserialize-fails.rs:3:28
  |
3 | #[derive(Debug, PartialEq, NetlinkAttributeDeserializable)]
  |                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  |
  = help: message: One variant must be marked with #[nla_type(_)] for wildcard handling. None found.
